hoot-core/src/main/cpp/hoot/core/io/OsmApiChangesetElement.h

```diff
@@ -50,22 +50,25 @@ class ElementIdToIdMap;
 
 /** Object that matches an XML tag to its XML attributes */
 typedef QPair<QString, QXmlStreamAttributes> XmlObject;
+typedef std::vector<std::pair<QString, QString>> ElementAttributes;
+typedef std::pair<QString, QString> ElementTag;
+typedef std::vector<ElementTag> ElementTags;
```

Comment:
This is where a lot of the memory was reduced, instead of keeping track of `QXmlStreamAttributes` objects only the string values are parsed out and kept around.